app/controllers/concerns/exception_hunter/authorization.rb

```diff
@@ -0,0 +1,25 @@
+module ExceptionHunter
+  module Authorization
+    extend ActiveSupport::Concern
+
+    included do
+      before_action :authenticate_admin_user_class
+    end
+
+    def authenticate_admin_user_class
+      admin_user_class = underscored_admin_user_class
+      return if admin_user_class.nil? ||
+                (admin_user_class && send("authenticate_#{admin_user_class}!"))
+
+      redirect_to_login
```

Comment:
This is not needed, just
```suggestion
      send("authenticate_#{underscored_admin_user_class}!") unless underscored_admin_user_class
```